Source/Storages/Cloud/CloudKeyManager.swift

```diff
@@ -89,8 +92,87 @@ internal class CloudKeyManager {
 
         return cloudKeyStorage
     }
+
+    private func pullKeyValue(named keyName: String, brainKeyPair: VirgilKeyPair) throws -> KeyknoxValue {
+        let pullParams = KeyknoxPullParams(identity: self.identity,
+                                           root: self.namedKeysRoot,
+                                           path: self.namedKeysPath,
+                                           key: keyName)
+
+        let keyknoxValue = try self.keyknoxManager
+            .pullValue(params: pullParams,
+                       publicKeys: [brainKeyPair.publicKey],
+                       privateKey: brainKeyPair.privateKey)
+            .startSync()
+            .get()
+
+        return keyknoxValue
+    }
+
+    private func pushKeyValue(named keyName: String, data: Data, hash: Data?, brainKeyPair: VirgilKeyPair) throws {
+        let pushParams = KeyknoxPushParams(identities: [self.identity],
+                                           root: self.namedKeysRoot,
+                                           path: self.namedKeysPath,
+                                           key: keyName)
+
+        _ = try self.keyknoxManager
+            .pushValue(params: pushParams,
+                       data: data,
+                       previousHash: hash,
+                       publicKeys: [brainKeyPair.publicKey],
+                       privateKey: brainKeyPair.privateKey)
+            .startSync()
+            .get()
+    }
+}
+
+extension CloudKeyManager {
```

Comment:
Maybe a better idea is to move optional parameter keyName to the underlying functions.